bson/json_util.py

```diff
@@ -634,6 +634,8 @@ def _parse_canonical_code(doc):
 def _parse_canonical_regex(doc):
     """Decode a JSON regex to bson.regex.Regex."""
     regex = doc['$regularExpression']
+    if not isinstance(regex['options'], str):
+        raise TypeError('Bad $regularExpression options, options must be string')
```

Comment:
Please move this check to below the len(doc) and len(regex) checks to ensure we give good error messages in all cases. For the same reason can you also add the type() of regex['options'] to the error message?

Reply:
Done.